util/metrics/import_metrics_files.rb

```diff
@@ -0,0 +1,181 @@
+    # The main entry point to the import_metrics_files script
+    #
+    # @author Bill Claytor
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files
+      pmc_dir = "#{@results_dir}/puppet-metrics-collector"
+      raise "Directory not found: #{pmc_dir}" unless File.directory? pmc_dir
+
+      puts "Importing puppet_metrics_collector files:"
+      puts " results_dir: #{@results_dir}"
+      puts " prefix: #{@prefix}"
+      puts " id: #{@id}"
+
+      service_dirs = Dir.glob("#{pmc_dir}/*").select { |f| File.directory? f }
+
+      puts " service directories:"
+      puts service_dirs
+      puts
+
+      # each service directory
+      service_dirs.each do |service_dir|
+        puts "Checking directory: #{service_dir}"
+
+        # each host directory
+        host_dirs = Dir.glob("#{service_dir}/*").select { |f| File.directory? f }
+
+        puts " host directories: "
+        puts host_dirs
+        puts
+
+        host_dirs.each do |host_dir|
+          import_metrics_files_for_host_dir(host_dir)
+        end
+      end
+    end
+
+    # Calls the json2graphite.rb script for the specified host directory
+    # with a server tag using the following pattern: prefix_id_hostname
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] host_dir The host directory to import
+    #
+    # @return [void]
+    #
+    # @example
+    #   import_metrics_files
+    #
+    def import_metrics_files_for_host_dir(host_dir)
+      hostname = File.basename host_dir
+      server_tag = "#{@prefix}_#{@id}_#{hostname}"
+      pattern = "'#{host_dir}/*.json'"
+      cmd = "ruby #{@json2graphite_path} --pattern #{pattern}" \
+        " --convert-to influxdb --netcat localhost --influx-db puppet_metrics --server-tag #{server_tag}"
+      puts "Importing puppet-metrics-collector files for host: #{hostname}"
+      puts " cmd: #{cmd}"
+      puts
+
+      `#{cmd}`
+    end
+  end
+end
+
+if $PROGRAM_NAME == __FILE__
+
+  DEFAULT_JSON2GRAPHITE_PATH = File.expand_path "~/git/puppet-metrics-viewer/json2graphite.rb"
+
+  DESCRIPTION = <<~DESCRIPTION
+    This script imports metrics collector data for the Standard and Large reference architectures.
+    It checks each service subdirectory of the puppet-metrics-collector directory and calls the json2graphite.rb script once for each host subdirectory found within.
+    The server tags are constructed using the following pattern:
+    prefix_id_hostname
+
+    For example:
+    be ruby util/metrics/import_metrics.rb -r results/scale/PERF_SCALE_12345 -p slv-649
+
+    This will result in the following server tags:
+    slv-649_12345_127.0.0.1
+    slv-649_12345_ip-10-227-0-22.amz-dev.puppet.net
+    slv-649_12345_ip-10-227-2-173.amz-dev.puppet.net
+
+  DESCRIPTION
+
+  DEFAULTS = <<~DEFAULTS
+
+    The following defaults values are used if the options are not specified:
+    * JSON2GRAPHITE_PATH (-j, --json2graphite): #{DEFAULT_JSON2GRAPHITE_PATH}
```

Comment:
will need update for the default results dir and the nil prefix if added.

Reply:
I'm not sure how a default results dir or nil prefix would work. There is a ticket to add an option to optionally suppress the ID portion of the server tag (SLV-685) in case you just want to use the prefix.

Reply:
Should be the same mech as the json...path.  Just set it to ".".  Could get fancy and call Dir.pwd.  Then let the user input override the default.
nil for the prefix could just be setting it to "" or to nil.  And a one line change to not use it when building the tag if nil or "".

Reply:
This seems error-prone and potentially confusing to users. Having the path to the json2graphite script set as a default makes sense since that is the primary use case. I think it is better to be explicit with the results dir to ensure that the user is actually running it against a results dir and that it is the one they mean to run against. The primary use case I see is the user running this script from the root of the GPLT repo and specifying the desired results dir (which is the use case described in the doc):
```
be ruby util/metrics/import_metrics_files.rb -r results/scale/PERF_SCALE_12345 -p slv-666
```

I doubt most users would navigate to the results dir and then try to run the script from there (which would require specifying the full path to the script or a relative path with "`../../` etc...".

Not specifying a prefix would result in a server tag which just contains the ID which does not seem useful (the prefix is what provides context to the run). When combined with the option to suppress the ID portion, having no server tag specified would likely result in an error from the json2graphite script.

Reply:
Having a default for a directory input be the current working dir is really common.  So I don't see much chance of it being confusing.  As for running in the wrong dir, they might run it in the gplt root, but that should bail out pretty fast since there is no puppet-metrics-collector dir there.  And I can't imagine a user being in one perf dir and wanting to run it on another.  But even if they did... it doesn't do any damage, just puts in additional data.  So the cost of the mistake is really low.
I personally run from the perf dir as I usually an looking at other files in there before I upload to influx...  For my first run in tmux session I cut and paste the command path from a helper file, after that I just use the command history and run the command again, just from a different dir... like this:
/home/randell/git/puppet-metrics-viewer/json2graphite.rb --pattern puppet-metrics-collector/**/*.json --convert-to influxdb --netcat localhost --influx-db puppet_metrics --server-tag new_8300_3
I just edit the last arg and off I go. Simple and fast.

On the prefix.  It seems bad to force a person to input a value that they may not have anything meaningful for.  Like say I don't have a meaningful prefix... why force me to add one.  But this is mostly a semantic.  As I personally will always use the prefix...  But it makes me think of all those "unknown_*" files in gplt root... it forced an input when there wasn't a meaningful one... so unknown was given.
That said, you will have to watch out for the case of no prefix and suppressed ID anyway, as the user can input "" as the prefix.  Thought actually I don't think you get an error, it just adds more data to that tag... and I think overwrites existing data if the timestamps line up...  that or ignores if there is already data of the same timestamp...  so maybe you don't have to care about it really.

Reply:
Updated the spec tests